core/embed/prodtest/optiga_prodtest.c

```diff
@@ -160,17 +175,24 @@ void pair_optiga(void) {
       optiga_pairing_state = OPTIGA_PAIRING_ERR_READ;
       return;
     }
+
+#ifdef STM32U5
+    secure_aes_ecb_decrypt_hw(secret, sizeof(secret), secret_dec,
+                              SECURE_AES_KEY_DHUK);
+#else
+    memcpy(secret_dec, secret, sizeof(secret_dec));
+#endif
```

Comment:
Code duplication. Let's read the secret using `secret_optiga_get()`.

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/3e52219453a68f9a6b3b35ef40029aa589a9f230